src/py42/exceptions.py

```diff
@@ -140,13 +142,28 @@ def __init__(self, exception, rule_id, source):
         )
 
 
+class Py42MFARequiredError(Py42UnauthorizedError):
+    """An exception raised when a request requires multi-factor authentication"""
+
+    def __init__(self, exception, message=None):
+        message = message or u"User requires multi-factor authentication."
+        super(Py42MFARequiredError, self).__init__(exception, message)
+
+
 def raise_py42_error(raised_error):
     """Raises the appropriate :class:`py42.exceptions.Py42HttpError` based on the given
     HTTPError's response status code.
     """
     if raised_error.response.status_code == 400:
         raise Py42BadRequestError(raised_error)
     elif raised_error.response.status_code == 401:
+        if raised_error.response.text:
+            response_dict = json.loads(raised_error.response.text)
+            if response_dict["error"][0]["primaryErrorKey"] in [
```

Comment:
I just realized that this will blow up if we ever get a 401 from anywhere that does not have this json structure. We need to make sure that this logic is nicely skipped if that happens. The easiest way is probably not just check for those error strings being _somewhere_ in the response (no need to use json.loads and check specific properties).

Reply:
Updated